### PR TITLE
Fix the Linux build and some style issues.

### PR DESCRIPTION
**Pull request by pgavlin**
- The PAL definitions we need from CoreCLR moved out of libcoreclr and
  into libcoreclrpal. Update CMakeLists accordingly.
- The JIT options sources were inconsistent with the rest of the code
  w.r.t. the use of the `this` pointer and were missing copyright
  headers.

### CODE REVIEW

**Comment by pgavlin** [comment]
@adiaaida @russellhadley PTAL

**Comment by mjsabby** [comment]
LGTM

**Comment by adiaaida** [comment]
LGTM

File: lib/Jit/options.cpp [hunk 1]
```diff
@@ -17,19 +33,19 @@ Options::Options(LLILCJitContext *Context) : Context(Context) {}
 
 void Options::initialize() {
   // Set 'IsAltJit' based on environment information.
-  this->setIsAltJit();
+  setIsAltJit();
```
Comment (russellhadley) [inline lib/Jit/options.cpp]: Is it *really* in the style guide to remove the "this"?  Not prepending members with 'this->' is like my a #1 pet peeve.

Comment (pgavlin) [inline lib/Jit/options.cpp]: Yes, it is. We are only using `this` when necessary for disambiguation.

**Pull request merged**